elasticsearch/docker/templates/Dockerfile.j2

```diff
@@ -71,17 +94,23 @@ RUN set -ex; \
  \
     && make \
  \
-    && git clone --branch v1.8.0.0 https://github.com/opendistro-for-elasticsearch/k-NN.git /usr/share/elasticsearch/k-NN \
+    && export KNN_VERSION=`git ls-remote --tags https://github.com/opendistro-for-elasticsearch/k-NN.git v* | grep {{version_tag}} | grep -oh "v[0-9.]*" | sort | tail -n 1` \
+ \
+    && git ls-remote --tags https://github.com/opendistro-for-elasticsearch/k-NN.git v* \
+ \
+    && echo KNN_VERSION $KNN_VERSION \
+ \
+    && git clone --branch $KNN_VERSION https://github.com/opendistro-for-elasticsearch/k-NN.git /usr/share/elasticsearch/k-NN \
  \
```

Comment:
These commands have been changed. I am working on it with Knn  team. will update it later.

Reply:
Ok, but this is a git clone, are you talking about the make or g++ commands?

Reply:
We were cloning specific branch because of [this](https://github.com/opendistro-for-elasticsearch/k-NN/tree/v1.8.0.0/jni/src) file which has been [removed](https://github.com/opendistro-for-elasticsearch/k-NN/tree/master/jni/src) now and another approach is being followed

Reply:
Thanks, waiting for your updates on this @weicongs-amazon

Reply:
We will move along with this change and update/optimize KNN process in a later PR. Thanks.